marlowe-dashboard-client/src/Marlowe/Execution/State.purs

```diff
@@ -1,133 +1,75 @@
+import Marlowe.Execution.Types (State, NamedAction(..), PendingTimeouts)
+import Marlowe.Semantics (Accounts, Action(..), Case(..), ChoiceId(..), Contract(..), Input, Party, ReduceResult(..), Slot(..), SlotInterval(..), Timeouts(..), Token, TransactionInput(..), TransactionOutput(..), _accounts, _boundValues, _minSlot, computeTransaction, emptyState, evalValue, makeEnvironment, reduceContractUntilQuiescent, timeouts)
+import Marlowe.Semantics (State) as Marlowe
 
--- This represents a previous step in the execution. The state property corresponds to the state before the
--- txInput was applied and it's saved as an early optimization to calculate the balances at each step.
-type PreviousState
-  = { txInput :: TransactionInput
-    , state :: State
-    }
-
--- This represents the timeouts that hasn't been applied to the contract. When a step of a contract has timeout
--- nothing happens until the next InputTransaction. This could be an IDeposit or IChoose for the continuation
--- contract, or an empty transaction to advance or even close the contract. We store a separate contract and state
--- than the "current" contract/state because this is the predicted state that we would be in if we applied an
--- empty transaction, and this allow us to extract the NamedActions that needs to be applied next.
--- Also, we store the timeouts as an Array because it is possible that multiple continuations have timeouted
--- before we advance the contract.
-type PendingTimeouts
-  = { timeouts :: Array Slot
-    , contract :: Contract
-    , state :: State
-    }
-
-type ExecutionState
-  = { previous :: Array PreviousState
-    , current ::
-        { state :: State
-        , contract :: Contract
-        }
-    , mPendingTimeouts :: Maybe PendingTimeouts
-    , mNextTimeout :: Maybe Slot
-    }
-
-_previousState :: Lens' ExecutionState (Array PreviousState)
-_previousState = prop (SProxy :: SProxy "previous")
-
-_previousTransactions :: Traversal' ExecutionState TransactionInput
-_previousTransactions = prop (SProxy :: SProxy "previous") <<< traversed <<< prop (SProxy :: SProxy "txInput")
-
-_currentState :: Lens' ExecutionState State
-_currentState = prop (SProxy :: SProxy "current") <<< prop (SProxy :: SProxy "state")
-
-_currentContract :: Lens' ExecutionState Contract
-_currentContract = prop (SProxy :: SProxy "current") <<< prop (SProxy :: SProxy "contract")
-
-_mNextTimeout :: Lens' ExecutionState (Maybe Slot)
-_mNextTimeout = prop (SProxy :: SProxy "mNextTimeout")
-
-_pendingTimeouts :: Traversal' ExecutionState (Array Slot)
-_pendingTimeouts = prop (SProxy :: SProxy "mPendingTimeouts") <<< _Just <<< prop (SProxy :: SProxy "timeouts")
-
-isClosed :: ExecutionState -> Boolean
-isClosed { current: { contract: Close } } = true
-
-isClosed _ = false
-
-initExecution :: Slot -> Contract -> ExecutionState
+initExecution :: Slot -> Contract -> State
```

Comment:
between the module changes (Execution -> State, Type, Lenses) and some of the renamings, this is a little hard to follow. I'd suggest taking the feedback of this PS and separate the changes into multiple ones.